### DISCUSSION TITLE
Prod ssr https quasar 2.6.6

### DISCUSSION THREAD

**Discussion by mirkofisic** [quasarframework/quasar#13287] [CLI - SSR mode] [answered]
How to start ssr production with https? Where to load certs? In development mode is very clear and it works but how to setup ssl for production?

**Comment by MilosPaunovic** [comment]
Maybe this could be of help:
[https://forum.quasar-framework.org/topic/2594/solved-how-do-you-setup-https-for-both-dev-and-prod-with-your-own-keys-and-certs](https://forum.quasar-framework.org/topic/2594/solved-how-do-you-setup-https-for-both-dev-and-prod-with-your-own-keys-and-certs)

**Reply by mirkofisic** [reply, marked as answer]
I used the production-export.js file and the ssrProductionExport function to load and set the certificate in this way and it works, I honestly don't know if this is correct?
```
import { ssrProductionExport } from 'quasar/wrappers'
import https from 'https'
import * as fs from 'fs'
const privateKey = fs.readFileSync('../../key.pem', 'utf8')
const certificate = fs.readFileSync('../../cert.pem', 'utf8')

export default ssrProductionExport(({ app, port, isReady }) => {
  return isReady().then(() => {
    const credentials = { key: privateKey, cert: certificate }
    const server = https.createServer(credentials, app)
    server.listen(port, () => {
      console.log('Server listening at port ' + port)
    })
  })
})
```
[Quasar Serve](https://quasar.dev/quasar-cli-vite/commands-list#serve) command is this only for SPA does SSR works with Serve?

**Reply by mirkofisic** [reply]
@rstoenescu guys can any of you from the development team confirm if this is ok?

**Reply by GregorioLopezO** [reply]
Your solution has worked for me. I've had to upgrade Quasar for using production-export.js.

Thank you, mirkofisic.

**Comment by operez2000** [comment]
@mirkofisic Thanks for your huge contribution, it works perfect! you save my day, there is lack of information about it. I don't know if that's correct as you question about but works perfect.
